core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -210,6 +210,16 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
     }
   }
 
+  def truncateCheckpoint(dataDir: File, topicAndPartition: TopicAndPartition, offset: Long) {
```

Comment:
@SinghAsDev I don't yet have a good idea of what the test look like? Could you provide some suggestion? If there is no easy way to test it, I will not be able to work on the test in a few days.